### PR TITLE
Code from Monday 27.05.

### PR DESCRIPTION
**Pull request by jovansimonovic**
Frontend code from monday's lectures. Please confirm PR

### CODE REVIEW

**Review by pbisevac — CHANGES_REQUESTED** [review]
Continue with this branch and add code from 29.05. 

We have alot of bugs in this branch, you can fix it with code from 10. labs.

**Review by jovansimonovic — COMMENTED** [review]
Understood. Will request re-review once I'm done

**Review by pbisevac — CHANGES_REQUESTED** [review]
Continue with this branch. We have to fix some bugs.

**Review by jovansimonovic — COMMENTED** [review]
Code from may 27th, 29th and 31st. Got a bit tangled up in some merge conflicts x), finally resolved them (at least I think I did :P). Praying to God the code still works :D

**Comment by pbisevac** [comment]
> Code from may 27th, 29th and 31st. Got a bit tangled up in some merge conflicts x), finally resolved them (at least I think I did :P). Praying to God the code still works :D

hehe, perfect practice 🥇

**Review by jovansimonovic — COMMENTED** [review]
App works now. Cleaned up the files left from all the chaos from conflicts. Is it safe to merge now?

File: src/hooks/useRefreshToken.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  const headers = {
+    authorization: `Bearer ${session?.user?.refreshToken}`,
+  };
+
+  const refreshToken = async () => {
+    const response = await post("/auth/refresh-token", {}, { headers });
+
+    if (session && response && response.status === 200 && response.data) {
+      await update({
```
Comment (pbisevacSing) [inline src/hooks/useRefreshToken.js:17]: We have a bug here. This must be changed. We can not update the session here.

File: src/hooks/useAuth.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        const prevRequest = error?.config;
+
+        if (
+          (error?.response?.status === 401 ||
+            error?.response?.status === 403) &&
+          !prevRequest?.sent
+        ) {
+          prevRequest.sent = true;
+
+          const refreshTokenResponse = await refreshToken();
```
Comment (pbisevacSing) [inline src/hooks/useAuth.js:34]: Here too, this is a bug. you must fix it. Try with another branch.

**Review by pbisevacSing — APPROVED** [review]
You can merge but dont forget to read comments.

File: src/hooks/useRefreshToken.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { post } from "@/core/httpClient";
+import { signIn, useSession } from "next-auth/react";
+
+export const useRefreshToken = () => {
+  const { data: session } = useSession();
```
Comment (pbisevacSing) [inline src/hooks/useRefreshToken.js:5]: missing update method here

**Review by pbisevacSing — APPROVED** [review]

**Pull request merged**